### PR TITLE
Remove HTTP library and add as a submodule pointer

### PR DESCRIPTION
**Pull request by leegeth**
*Issue #, if available:*

*Description of changes:*
Remove HTTP library and add as a submodule pointer
https://github.com/FreeRTOS/coreHTTP is added as a submodule in libraries/standard/coreHTTP.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: demos/http/http_demo_basic_tls/CMakeLists.txt [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
+
 # Demo target.
 add_executable(${DEMO_NAME})
 
 target_sources(
     ${DEMO_NAME}
     PRIVATE
         "${DEMO_NAME}.c"
+        ${HTTP_SOURCES}
+        "${HTTP_INCLUDE_THIRD_PARTY_DIRS}/http_parser.c"
```
Comment (sarenameas) [inline demos/http/http_demo_basic_tls/CMakeLists.txt:14]: I have merged you may change this to ${HTTP_THIRD_PARTY_SOURCES}

Comment (leegeth) [reply demos/http/http_demo_basic_tls/CMakeLists.txt:14]: Thanks.

File: .gitmodules [hunk 2]
```diff
@@ -9,4 +9,7 @@
 	url = https://github.com/awslabs/aws-templates-for-cbmc-proofs.git
 [submodule "libraries/standard/coreMQTT"]
 	path = libraries/standard/coreMQTT
-	url = https://github.com/FreeRTOS/coreMQTT.git
+	url = git@github.com:FreeRTOS/coreMQTT.git
```
Comment (nateglims) [inline .gitmodules:12]: Please remove this change.

Comment (leegeth) [reply .gitmodules:12]: ok. Thanks.

**Review by sarenameas — APPROVED** [review]

**Review by aggarg — DISMISSED** [review]

**Review by sarenameas — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**